sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -344,8 +355,21 @@ private void checkPushInIntent(Intent intent) {
      *
      * @param remoteMessage push message details.
      */
-    synchronized void onMessageReceived(RemoteMessage remoteMessage) {
+    void onMessageReceived(final RemoteMessage remoteMessage) {
         MobileCenterLog.info(LOG_TAG, "Received push message in foreground id=" + remoteMessage.getMessageId());
+        postOnUiThread(new Runnable() {
+
+            @Override
+            public void run() {
+                handleOnMessageReceived(remoteMessage);
+            }
+        });
+    }
+
+    /**
+     * Top level method needed for synchronized code coverage.
+     */
+    private synchronized void handleOnMessageReceived(RemoteMessage remoteMessage) {
```

Comment:
I had to look thru method callers to figure out which thread the methods will run.
Can you just add ui thread annotation here so that readers can easily understand?